src/site/content/en/progressive-web-apps/customize-install/index.md

```diff
@@ -0,0 +1,181 @@
+2. Save the `beforeinstallprompt` event, so it can be used to trigger the
+   install flow later.
+3. Alert the user your PWA is installable, and provide a button or other
+   element to start the in-app installation flow.
+
+### Listen for the beforeinstallprompt event
+
+If your Progressive Web App meets the required [installation criteria](/install-criteria/),
+the browser fires a `beforeinstallprompt` event. Save a reference to the
+event, and update your user interface to indicate that the user can install
+your PWA. This is highlighted below.
+
+```js/5-8
+let deferredPrompt;
+
+window.addEventListener('beforeinstallprompt', (e) => {
+  // Prevent the mini-infobar from appearing on mobile
+  e.preventDefault();
+  // Stash the event so it can be triggered later.
+  deferredPrompt = e;
+  // Update UI notify the user they can install the PWA
+  showInstallPromotion();
+});
+```
+
+There are many different [patterns](/install-patterns/) that you can use to
+notify the user your app can be installed and provide an in-app install
+flow, for example, a button in the header, an item in the navigation menu,
+or an item in your content feed.
+
+{% Aside 'caution' %}
+You may want to wait before showing the prompt to the user so you don't
+distract them from what they're doing. For example, if the user is in a
+check-out flow, or creating their account, let them complete that before
+interrupting them with the prompt.
+{% endAside %}
+
+### In-app installation flow
+
+To provide an in-app installation flow, you'll need to provide a button,
+or other user interface element within your app that the user can click,
+indicating users can install your Progressive Web app. When the element is
+clicked, call `prompt()` on the saved `beforeinstallprompt` event (stored
+in the `deferredPrompt` variable). It shows the user a modal install dialog,
+asking them to confirm they want to install your PWA.
+
+```js
+buttonInstall.addEventListener('click', (e) => {
+  // Hide the install promotion
+  hideInstallPromotion();
+  // Show the install prompt
+  deferredPrompt.prompt();
+  // Wait for the user to respond to the prompt
+  deferredPrompt.userChoice.then((choiceResult) =>{
+    if (choiceResult.outcome === 'accepted') {
+      console.log('User accepted the install prompt');
+    } else {
+      console.log('User dismissed the install prompt');
+    }
+  })
+});
+```
+
+The `userChoice` property is a promise that resolves with the user's choice.
+You can only call `prompt()` on the deferred event once. If the user
+dismisses it, you'll need to wait until the `beforeinstallprompt` event
+is fired again, typically immediately after the `userChoice` property
+has resolved.
+
+## Detect when the PWA was successfully installed
+
+You can use the `userChoice` property to determine if the user installed
+your app from within your user interface. But, if the user installs your
+PWA from the address bar, or other browser component, `userChoice` won't
```

Comment:
'Or' only needs a comma when:

* used as a coordinating conjunction between two independent clauses (clauses that could be sentences in their own right).
* separates the final two items in a list of three or more. (You only have two items here.)

Reply:
fixed!